src/objects/Contact.object

```diff
@@ -3671,6 +3671,20 @@
         <trackFeedHistory>false</trackFeedHistory>
         <type>Lookup</type>
     </fields>
+    <fields>
+        <fullName>Primary_Language__c</fullName>
+        <deleteConstraint>SetNull</deleteConstraint>
+        <description>The language the contact prefers to use.</description>
+        <externalId>false</externalId>
+        <inlineHelpText>The language the contact prefers to use.</inlineHelpText>
+        <label>Primary Language</label>
+        <referenceTo>Contact_Language__c</referenceTo>
```

Comment:
@robertbwatson -- just a quick question here -- should this be a reference to the "Contact Language" or the actual Language base object? I ask because the reference is circular, Contact_Language looks up to Contact which has a lookup to Contact Language.  if we populate the Language__c in the lookup instead can potentially go to a Language object and view its related list and see all the contacts who have that language set as primary. I think this is what I had in mind originally. Does that break things? I feel it doesn't but feel free to yell at me :)

Reply:
@lifewithryan That shouldn't break things and probably makes more sense. This will get around the issue of potentially selecting a Contact Language record that isn't even related to the contact.

I would argue then that this should still not be on the page layout by default, and if it's added, we should recommend that it is read-only (so that the custom code sets it when creating the Contact Langauge record)

The spec has it as Contact Language so I'll be sure to update that, and we'll need to update the ERD.

Reply:
"view its related list and see all the contacts who have that language set as primary." -- Since we have the Contact Language related list on the Language page layout already, which shows the contact, fluency, and if it's the primary language, I think we're okay here and probably don't need to add the additional related list on the Language object. But the lookup is still good to have.

Reply:
agreed, at this point, having the related list would just be a "convenience" sort of thing. Largely unusable in sizeable orgs...I think I'm just used to working with smaller footprints :)

Reply:
With respect to having the field be Read Only. I think that is okay but, maybe we should consider that if someone sets that field, we would in turn create (if it didn't exist) a Contact Language Record for that language on the contact and set the Primary checkbox to true.

Reply:
I think that could maybe be a future work item/improvement. I'm not sure if we want to handle that now. Would we assume that the Fluency would be "Fluent"? (Let the user decide with a new custom setting?) etc

It might be worthwhile to have now, in which case I could ask if it makes sense to do in W-031856 or a separate WI.

Reply:
If it's my primary language it better be fluent, but that's a good point...